pdns/recursor_cache.cc

```diff
@@ -145,11 +144,11 @@ time_t MemRecursorCache::handleHit(MapCombo::LockedContent& content, MemRecursor
   time_t ttd = entry->d_ttd;
   origTTL = entry->d_orig_ttl;
 
-  if (variable && (!entry->d_netmask.empty() || entry->d_rtag)) {
+  if (variable != nullptr && (!entry->d_netmask.empty() || entry->d_rtag)) {
     *variable = true;
   }
 
-  if (res) {
+  if (!res->empty()) {
```

Comment:
This is wrong, `res` can be a `nullptr`

Reply:
Oh good spot, totally true. With the related issue I close about .yy files and me not forgetting a make I missed this one Should have put this PR in Draft.